cmake/FindLuaJit.cmake

```diff
@@ -32,8 +32,11 @@ if(LUAJIT_USE_STATIC)
 endif()
 
 list(APPEND LUAJIT_NAMES luajit-5.1)
+if(MSVC)
```

Comment:
No. @equalsraf's branch compiles on MinGW too which doesn't need this one.

See equalsraf@bcd0579.

To be honest I should probably squash equalsraf@94dcbe0, and equalsraf@bcd0579
into this:

```
if(MSVC)
  list(APPEND LUAJIT_NAMES lua51)
elseif(MINGW)
  list(APPEND LUAJIT_NAMES libluajit)
else()
  list(APPEND LUAJIT_NAMES luajit-5.1)
endif()
```

Further for consistency with the rest of the build system if we want to check for Windows we do this: `if(WIN32)`.

@justinmk in response to your earlier question I'm not sure why the library names are different but you can see that `third-party/cmake/BuildLuajit.cmake` and `third-party/cmake/BuildLibuv.cmake` is also using those library names. The MinGW support for `luajit` is in equalsraf@b95df5e.